exercises/javascript-view-swapping/main.js

```diff
@@ -0,0 +1,25 @@
+var $tabContainer = document.querySelector('.tab-container');
+var $tab = document.querySelectorAll('.tab');
+var $view = document.querySelectorAll('.view');
+$tabContainer.addEventListener('click', matches);
+function matches(e) {
+  if (e.target.matches('.tab')) {
+    for (var i = 0; i < $tab.length; i++) {
+      if ($tab[i] !== e.target) {
+        $tab[i].className = 'tab';
+      } else {
+        $tab[i].className = 'tab active';
+      }
+    }
+  }
+  if (e.target.matches('.tab')) {
```

Comment:
This would be a great time to use a guard. Rather than only doing things if the target has the `tab` class, you should check at the beginning if the element doesn't have a `tab` class, and if it doesn't, `return` out of the function.